src/Servers/Kestrel/Core/src/Internal/Infrastructure/ConnectionManager.cs

```diff
@@ -3,11 +3,14 @@
 
 using System;
 using System.Collections.Concurrent;
+using System.Threading;
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure
 {
     internal class ConnectionManager
     {
+        private static long _lastConnectionId = long.MinValue;
```

Comment:
Only one instance of ConnectionManager is created. Change to not be static?

Reply:
Yes please

Reply:
Consider that I could start two Host instances in the same process, we shouldn't have any static state that would break that.